lib/file_validators/utils/content_type_detector.rb

```diff
@@ -1,44 +1,85 @@
-begin
-  require 'cocaine'
-rescue LoadError
-end
-
 module FileValidators
   module Utils
 
     class ContentTypeDetector
-      EMPTY_CONTENT_TYPE = 'inode/x-empty'
-      DEFAULT_CONTENT_TYPE = 'application/octet-stream'
+      # The content-type detection strategy is as follows:
+      #
+      # 1. Blank/Empty files: If there's no filepath or the file is empty,
+      #    provide a sensible default (application/octet-stream or inode/x-empty)
+      #
+      # 2. Calculated match: Return the first result that is found by both the
+      #    `file` command and MIME::Types.
+      #
+      # 3. Standard types: Return the first standard (without an x- prefix) entry
+      #    in MIME::Types
+      #
+      # 4. Experimental types: If there were no standard types in MIME::Types
+      #    list, try to return the first experimental one
+      #
+      # 5. Raw `file` command: Just use the output of the `file` command raw, or
+      #    a sensible default. This is cached from Step 2.
 
-      def initialize(file_path)
-        @file_path = file_path
-      end
+      EMPTY_TYPE = "inode/x-empty"
+      SENSIBLE_DEFAULT = "application/octet-stream"
 
-      # content type detection strategy:
-      #
-      # 1. empty file: returns 'inode/x-empty'
-      # 2. nonempty file: if the file is not empty then returns the content type using file command
-      # 3. invalid file: file command raises error and returns 'application/octet-stream'
+      def initialize(filepath)
+        @filepath = filepath
+      end
 
+      # Returns a String describing the file's content type
       def detect
-        empty_file? ? EMPTY_CONTENT_TYPE : content_type_from_file_command
+        if blank_name?
+          SENSIBLE_DEFAULT
```

Comment:
This feels a little weird - if we've been passed an empty @filepath then there is no file to give a type for. It sort of feels like we should return nil or throw an error rather than make up an answer.